Library/Formula/citus.rb

```diff
@@ -0,0 +1,55 @@
+class Citus < Formula
+  desc "PostgreSQL-based distributed RDBMS"
+  homepage "https://www.citusdata.com"
+  url "https://github.com/citusdata/citus/archive/v5.0.0.tar.gz"
+  sha256 "a72bd7e9020c11f19d08e58f1f8aa8e83e7f1f377facb6c8020fcaa917f9a3ee"
+
+  head "https://github.com/citusdata/citus.git"
+
+  depends_on "postgresql"
+
+  def install
+    config_args = %W[--prefix=#{prefix} PG_CONFIG=#{Formula["postgresql"].opt_bin}/pg_config]
+
+    # workaround for https://github.com/Homebrew/homebrew/issues/49948
+    make_args = ["libpq=-L#{Formula["postgresql"].opt_lib} -lpq"]
+
+    system "./configure", *config_args
+    system "make", *make_args
+
+    mkdir "stage"
+    system "make", "install", "DESTDIR=#{buildpath}/stage"
```

Comment:
So here's the deal:

Even though we have a `configure` script, `make install` doesn't really honor the `prefix` (which is why I removed it: it was misleading).

Instead, the PostgreSQL extension building support (documented [here](http://www.postgresql.org/docs/9.5/static/extend-pgxs.html)) will call `pg_config` to determine the bin, include, lib, and share directories known to the PostgreSQL installation.

Unfortunately, some of these are top-level (as in `/usr/local/lib`) and most others are buried within the PostgreSQL package's own Cellar (i.e. underneath `/usr/local/Cellar/postgresql/9.5.2/`).

_Even_ with a `DESTDIR` argument, PGXS will recreate this hierarchy underneath the destdir, which is the reason we need that glob. The binaries are actually at `$DESTDIR/usr/local/Cellar/postgresql/9.5.2/bin`.

I experimented with trying to force PGXS to just install the files directly within the package's Cellar, but its integration with `pg_config` is just too tight. As _all_ other PostgreSQL extension Homebrew formulas built using PGXS (PostGIS, pldebugger, pgtap, pgrouting, and pgroonga) have a `mkdir 'stage'` line, I think it would be _more_ confusing to have this formula try to deviate from the norm they establish.

So essentially the answer is "PostgreSQL has an extension building infrastructure that bypasses everything and installs files directly into the directories reported by `pg_config`, so we first install to a `DESTDIR` and then map that hierarchy to something more Homebrew-friendly".

FWIW, the RPM and Debian builds I authored had to do much the same thing: build the extension, install to a `DESTDIR`, grab the files out of there into a structure `yum` or `apt` understand, then create the `rpm` or `deb` package.